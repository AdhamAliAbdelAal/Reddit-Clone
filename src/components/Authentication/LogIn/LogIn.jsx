import { useEffect, useState } from 'react';

// mui components
import { Typography } from '@mui/material';

// componenets
import LoadingPage from '../LoadingPage/LoadingPage';
import AuthenticationHeader from '../AuthenticationHeader/AuthenticationHeader';
import ThirdParty from '../ThirdParty/ThirdParty';
import Divider from '../Divider/Divider';
import FirstParty from '../FirstParty/FirstParty';

// styles
import { AuthenticationBody, StyledLink } from '../styles';

/**
 * Login Page Component
 * @returns {React.Component} - Main Body of Login Page
 */
function LogIn() {
  // state
  const [remeberMe, setRemeberMe] = useState(false);

  // effect
  useEffect(() => {
    //= =>Check on Cookies
    setRemeberMe(false);
  }, []);

  const caption = (
    <>
      By continuing, you agree to our
      {' '}
      <StyledLink fontWeight="400" href="https://www.redditinc.com/policies/user-agreement" capital="none">User Agreement</StyledLink>
      {' '}
      and
      {' '}
      <StyledLink fontWeight="400" href="https://www.reddit.com/policies/privacy-policy" capital="none">Privacy Policy </StyledLink>
      .
    </>
  );

  return (
    <AuthenticationBody mnwidth="290px" mxwidth={remeberMe ? '440px' : '290px'} data-testid="login-test">
      {remeberMe ? <LoadingPage /> : (
        <>
          <AuthenticationHeader reddit={false} title="Log in" caption={caption} />
          <ThirdParty />
          <Divider />
          <FirstParty />
          <Typography paragraph fontSize="12px" fontFamily="ibm-plex-sans,sans-serif">
            Forgot your
            {' '}
            <StyledLink href="/username" fontWeight="400" capital="none">username</StyledLink>
            {' or '}
            <StyledLink href="/password" fontWeight="400" capital="none">password</StyledLink>
            {' ?'}
          </Typography>
          <Typography paragraph fontSize={12}>
            New to Reddit?
            {' '}
            <StyledLink href="/register" fontWeight="600" capital="uppercase">sign up</StyledLink>
          </Typography>
        </>
      )}
    </AuthenticationBody>
  );
}

export default LogIn;
