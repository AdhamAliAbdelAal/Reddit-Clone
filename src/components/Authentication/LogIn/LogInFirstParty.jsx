import { Typography } from '@mui/material';
import {
  FirstPartyContainer, AuthenticationInput, AuthenticationButton, StyledLink,
} from '../AuthenticationConatiners/styles';

function LogInFirstParty() {
  const LogIn = () => {
    // Check nonempty inputfileds
    console.log('login');
    /* BackAPI */
    // 1. Valid input username must be from 3-20 characters
    // 2.Note on login a message appears at the bottom till loading
  };
  return (
    <FirstPartyContainer width="280px" onSubmit={LogIn}>
      <AuthenticationInput label="USERNAME" variant="outlined" />
      <AuthenticationInput label="PASSWORD" variant="outlined" />

      <AuthenticationButton type="submit">Log In</AuthenticationButton>
      <Typography paragraph fontSize="12px" fontFamily="ibm-plex-sans,sans-serif">
        Forgot your
        {' '}
        <StyledLink href="/username" fontWeight="400" capital="none">username</StyledLink>
        {' or '}
        <StyledLink href="/password" fontWeight="400" capital="none">password</StyledLink>
        {' ?'}
      </Typography>

      <Typography paragraph fontSize={12}>
        New to Reddit?
        {' '}
        <StyledLink href="/register" fontWeight="600" capital="uppercase">sign up</StyledLink>
      </Typography>
    </FirstPartyContainer>
  );
}

export default LogInFirstParty;
