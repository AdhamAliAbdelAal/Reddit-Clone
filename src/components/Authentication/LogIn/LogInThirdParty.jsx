import { ThirdPartyContainer, ThirdPartyButton, ButtonIcon } from '../AuthenticationConatiners/styles';
import Google from '../../../assets/images/google.png';
import Facebook from '../../../assets/images/facebook.png';

// My be Same as SignUp If so reuse the Signup Thirs Party Compoonenet
function LogInThirdParty() {
  return (
    <ThirdPartyContainer width="280px">
      <ThirdPartyButton variant="outlined">
        <ButtonIcon src={Google} alt="Google" />
        Continue with Google
      </ThirdPartyButton>
      <ThirdPartyButton variant="outlined">
        <ButtonIcon src={Facebook} alt="Facebook" />
        Continue with Facebook
      </ThirdPartyButton>
    </ThirdPartyContainer>
  );
}

export default LogInThirdParty;
