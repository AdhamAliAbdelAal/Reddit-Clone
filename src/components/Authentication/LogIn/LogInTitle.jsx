import { Typography, Link } from '@mui/material';
import { AuthenticationTitle, AuthenticationHeader } from '../AuthenticationConatiners/styles';

function LogInTitle() {
  return (
    <AuthenticationTitle height="100px" width="100%" mnwidth="280px">
      <AuthenticationHeader variant="h1">Log in</AuthenticationHeader>
      <Typography fontSize="12px" fontFamily="Noto Sans,sans-serif">
        By continuing, you agree to our
        {' '}
        <Link href="https://www.redditinc.com/policies/user-agreement" underline="none">User Agreement</Link>
        {' '}
        and
        {' '}
        <Link href="https://www.reddit.com/policies/privacy-policy" underline="none">Privacy Policy</Link>
        .
      </Typography>
    </AuthenticationTitle>
  );
}

export default LogInTitle;
