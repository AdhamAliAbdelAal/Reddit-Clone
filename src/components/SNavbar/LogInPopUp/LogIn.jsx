/* eslint-disable import/no-cycle */
import { Typography, Box } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import * as React from 'react';
import StyledDialog from '../SignUpPopUp/styles';
import { StyledLink } from '../ِAuthentication/styles';
import AuthenticationHeader from '../../Authentication/AuthenticationHeader/AuthenticationHeader';
import ThirdParty from '../ِAuthentication/ThirdParty/ThirdParty';
import Divider from '../../Authentication/Divider/Divider';
import FirstParty from '../ِAuthentication/FirstParty/FirstParty';
import { LoginContext } from '../SNavbar';

/**
 *  LogIn popUp
 * @component
 * Login popup Differenet Components
 * @returns {React.Component} - Main Body of Login Page
 */
function LogIn() {
  const caption = (
    <>
      By continuing, you agree to our
      {' '}
      <StyledLink fontWeight="400" href="https://www.redditinc.com/policies/user-agreement" capital="none">User Agreement</StyledLink>
      {' '}
      and
      {' '}
      <StyledLink fontWeight="400" href="https://www.reddit.com/policies/privacy-policy" capital="none">Privacy Policy </StyledLink>
      .
    </>
  );

  const {
    openLogIn,
    handleClose,
    handleClickOpenForgotUser,
    handleClickOpenForgotPass,
    handleClickOpenSignUp,
  } = React.useContext(LoginContext);

  return (
    <StyledDialog
      data-testid="login-popup"
      fullscreen
      open={openLogIn}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <Box width="280px" height="454px">
        <CloseIcon
          sx={{
            position: 'absolute', right: '16px', top: '16px', cursor: 'pointer',
          }}
          onClick={handleClose}
        />
        <AuthenticationHeader reddit={false} title="Log in" caption={caption} />
        <ThirdParty circular={false} />
        <Divider />
        <FirstParty />
        <Typography paragraph fontSize="12px" fontFamily="ibm-plex-sans,sans-serif">
          Forgot your
          {' '}
          <StyledLink onClick={handleClickOpenForgotUser} data-testid="login-forgetuser" fontWeight="400" capital="none">username</StyledLink>
          {' or '}
          <StyledLink onClick={handleClickOpenForgotPass} data-testid="login-forgetpass" fontWeight="400" capital="none">password</StyledLink>
          {' ?'}
        </Typography>

        <Typography paragraph fontSize={12}>
          New to Reddit?
          {' '}
          <StyledLink onClick={handleClickOpenSignUp} data-testid="login-signup" fontWeight="600" capital="uppercase">sign up</StyledLink>
        </Typography>
      </Box>
    </StyledDialog>
  );
}

export default LogIn;
