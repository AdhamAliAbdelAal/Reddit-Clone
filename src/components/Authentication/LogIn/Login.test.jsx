import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import LogIn from './LogIn';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<LogIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
