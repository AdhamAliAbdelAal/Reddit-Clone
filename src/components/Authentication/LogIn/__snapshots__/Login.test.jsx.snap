// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-14rl5xe"
  data-testid="login-test"
  mnwidth="290px"
  mxwidth="290px"
>
  <div
    className="MuiBox-root css-eyuwdo"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-1sg842v-MuiTypography-root"
    >
      Log in
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-68wmct-MuiTypography-root"
    >
      By continuing, you agree to our
       
      <a
        capital="none"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
        href="https://www.redditinc.com/policies/user-agreement"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        User Agreement
      </a>
       
      and
       
      <a
        capital="none"
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
        href="https://www.reddit.com/policies/privacy-policy"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Privacy Policy 
      </a>
      .
    </p>
  </div>
  <div
    className="MuiBox-root css-1vt6mgo"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1inhtju-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="Google"
        className="css-1x4dpap"
        src="google.png"
      />
      continue with google
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1inhtju-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="Facebook"
        className="css-1x4dpap"
        src="facebook.png"
      />
      continue with facebook
    </button>
  </div>
  <div
    className="MuiBox-root css-1rocah8"
  >
    <hr
      className="css-1tv3lka"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-ikhzr9-MuiTypography-root"
    >
      OR
    </p>
    <hr
      className="css-1tv3lka"
    />
  </div>
  <form
    className="css-bs3vrj"
    data-testid="FirstParty-test"
    onSubmit={[Function]}
    width="290px"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-fp4mq4-MuiFormControl-root-MuiTextField-root"
      clr="#978A8C"
      data-testid="UserName-FirstParty-test"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root css-fp4mq4-MuiFormControl-root-MuiTextField-root"
      clr="#978A8C"
      data-testid="Password-FirstParty-test"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled css-1mog9k0-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r1:"
        id=":r1:-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
          disabled={false}
          id=":r1:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15fwilh-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      data-testid="login-btn-test"
      disabled={false}
      id=":r2:"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Log In
    </button>
  </form>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-1qvb19m-MuiTypography-root"
  >
    Forgot your
     
    <a
      capital="none"
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
      href="/username"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      username
    </a>
     or 
    <a
      capital="none"
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-y1krgh-MuiTypography-root-MuiLink-root"
      href="/password"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      password
    </a>
     ?
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-4owvyy-MuiTypography-root"
  >
    New to Reddit?
     
    <a
      capital="uppercase"
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xn5q3i-MuiTypography-root-MuiLink-root"
      href="/register"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      sign up
    </a>
  </p>
</div>
`;
